subsys/net/ip/connection.c

```diff
@@ -114,32 +114,42 @@ void conn_register_debug(struct net_conn *conn,
 #define conn_register_debug(...)
 #endif /* (CONFIG_NET_CONN_LOG_LEVEL >= LOG_LEVEL_DBG) */
 
+static K_MUTEX_DEFINE(conn_lock);
```

Comment:
I think it's worth making it enabled only if SMP is enabled. Just have macro to use in function below, to lock/unlock.
Sure, it's one little object, but on small targets: every bytes count.

Reply:
I think we have similar lock without SMP guards, like
```
./ip/route.c:47:static K_MUTEX_DEFINE(lock);
./ip/net_mgmt.c:37:static K_MUTEX_DEFINE(net_mgmt_lock);
./ip/dhcpv4.c:35:static K_MUTEX_DEFINE(lock);
./ip/tcp.c:57:static K_MUTEX_DEFINE(tcp_lock);
./ip/net_if.c:42:static K_MUTEX_DEFINE(lock);
```